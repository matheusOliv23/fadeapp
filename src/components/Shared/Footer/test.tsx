import { render, screen } from '@testing-library/react'

import { ButtonSocialProps } from 'components/Buttons/ButtonSocial'

import { Footer } from '.'

const { getByRole } = screen

const data: ButtonSocialProps[] = [
  {
    name: 'Instagram',
    urlIcon: 'https://res.cloudinary.com/user/icon.svg',
    urlRedirect: 'https://site.com/',
  },
  {
    name: 'Youtube',
    urlIcon: 'https://res.cloudinary.com/user/icon2.svg',
    urlRedirect: 'https://site2.com/',
  },
  {
    name: 'Tiktok',
    urlIcon: 'https://res.cloudinary.com/user/icon3.svg',
    urlRedirect: 'https://site3.com/',
  },
]

describe('<Footer />', () => {
  it('should render by default', () => {
    render(
      <Footer
        items={data}
        title="Title"
        company="Company"
        linkCompany="https://company.com/"
      />,
    )

    const title = getByRole('heading', { name: 'Title' })
    const instagram = getByRole('link', { name: 'Instagram' })
    const youtube = getByRole('link', { name: 'Youtube' })
    const tiktok = getByRole('link', { name: 'Tiktok' })
    const copyright = getByRole('link', { name: 'Company' })

    expect(title).toBeInTheDocument()
    expect(instagram).toBeInTheDocument()
    expect(youtube).toBeInTheDocument()
    expect(tiktok).toBeInTheDocument()
    expect(copyright).toHaveAttribute('href', 'https://company.com/')
  })

  it('Should match snapshot', () => {
    const { container } = render(
      <Footer
        items={data}
        title="Title"
        company="Company"
        linkCompany="https://company.com/"
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
