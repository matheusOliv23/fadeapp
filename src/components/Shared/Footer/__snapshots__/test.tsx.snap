// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should match snapshot 1`] = `
<div>
  <footer
    class="
flex
flex-col
justify-start
items-center
"
  >
    <h2
      class="
mb-8
text-2xl
font-medium
text-white
"
    >
      Title
    </h2>
    <div
      class="
mb-[4.5rem] 
flex w-full 
items-center 
justify-center 
gap-[3.4375rem]
"
    >
      <div
        class="
flex 
flex-col 
items-center 
gap-4
"
      >
        <a
          class="
relative
flex 
h-16 
w-16 
items-center 
justify-center 
rounded-full 
bg-primary 

transition-all
duration-300
ease-in-out

hover:scale-105

before:absolute
before:rounded-full
before:content-['']
before:w-0
before:h-0
before:transition-all
before:ease-in-out
before:duration-300

before:bg-black/30

hover:before:w-full
hover:before:h-full
hover:before:border-2
hover:before:border-primary
"
          href="https://site.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Instagram"
            src="https://res.cloudinary.com/user/icon.svg"
          />
        </a>
        <span
          class="
text-sm 
font-medium 
capitalize 
text-white
"
        >
          Instagram
        </span>
      </div>
      <div
        class="
flex 
flex-col 
items-center 
gap-4
"
      >
        <a
          class="
relative
flex 
h-16 
w-16 
items-center 
justify-center 
rounded-full 
bg-primary 

transition-all
duration-300
ease-in-out

hover:scale-105

before:absolute
before:rounded-full
before:content-['']
before:w-0
before:h-0
before:transition-all
before:ease-in-out
before:duration-300

before:bg-black/30

hover:before:w-full
hover:before:h-full
hover:before:border-2
hover:before:border-primary
"
          href="https://site2.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Youtube"
            src="https://res.cloudinary.com/user/icon2.svg"
          />
        </a>
        <span
          class="
text-sm 
font-medium 
capitalize 
text-white
"
        >
          Youtube
        </span>
      </div>
      <div
        class="
flex 
flex-col 
items-center 
gap-4
"
      >
        <a
          class="
relative
flex 
h-16 
w-16 
items-center 
justify-center 
rounded-full 
bg-primary 

transition-all
duration-300
ease-in-out

hover:scale-105

before:absolute
before:rounded-full
before:content-['']
before:w-0
before:h-0
before:transition-all
before:ease-in-out
before:duration-300

before:bg-black/30

hover:before:w-full
hover:before:h-full
hover:before:border-2
hover:before:border-primary
"
          href="https://site3.com/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Tiktok"
            src="https://res.cloudinary.com/user/icon3.svg"
          />
        </a>
        <span
          class="
text-sm 
font-medium 
capitalize 
text-white
"
        >
          Tiktok
        </span>
      </div>
    </div>
    <span
      class="
text-xs 
text-gray-700
font-normal
"
    >
      Â©
       
      <a
        class="
hover:text-primary
"
        href="https://company.com/"
        rel="noreferrer"
        target="_blank"
      >
        Company
      </a>
       
      - 
      2022
    </span>
  </footer>
</div>
`;
